cmake/GtsamBuildTypes.cmake

```diff
@@ -138,16 +138,22 @@ else()
   set(GTSAM_COMPILE_OPTIONS_PRIVATE_TIMING          -g -O3  CACHE STRING "(User editable) Private compiler flags for Timing configuration.")
 endif()
 
-# Enable C++11:
+# Enable C++17:
 if (NOT CMAKE_VERSION VERSION_LESS 3.8)
-    set(GTSAM_COMPILE_FEATURES_PUBLIC "cxx_std_11" CACHE STRING "CMake compile features property for all gtsam targets.")
+    set(GTSAM_COMPILE_FEATURES_PUBLIC "cxx_std_17" CACHE STRING "CMake compile features property for all gtsam targets.")
     # See: https://cmake.org/cmake/help/latest/prop_tgt/CXX_EXTENSIONS.html
-    # This is to enable -std=c++11 instead of -std=g++11
+    # TODO(dellaert): is following line still needed or was that only for c++11?
     set(CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
Yes, it's better to keep it to make compilation not to depend on compiler-specific features... so it's less likely that it builds ok on gcc and not on clang or the other way around.